TUTORIAL.md

```diff
@@ -0,0 +1,635 @@
+
+
+```
+  class Student
+    attr_accessor :name, :id, :tagline, :blog_url, :twitter, :image_url, :biography, :github
+
+    def self.create_table
+      sql = "CREATE TABLE students
+        (
+          id INTEGER primary key autoincrement,
+          name TEXT,
+          tagline TEXT,
+          blog_url TEXT,
+          twiiter TEXT,
+          image_url TEXT,
+          biography TEXT,
+        github TEXT
+       )"
+     DB[:conn].execute(sql)
+    end
+
+  end
+```
+Unfortunately the above won't pass the test. Can you guess why?
+
+> Because although we have the correct SQL command, we haven't executed the correct SQL command in our student database.
+
+Take a look at line 4 in `school-domain-with-db/config/environment` and notice that Line 4 is where we establish a connection with a sqlite Relational Database Management System.  To connect to this database in our `create_table` class method, we simply need to reference the `DB` constant and the `:conn` key. So basically `DB[:conn]` represents our applications' connection to the database.
+
+Next we need to execute the SQL command within the context of that database connection.  Sqlite3 has an `execute` method that accepts a string argument, which will be the string version of the SQL command we first wrote down. Our new `create_table` class method should now look like this:
+
+```
+  def self.create_table
+    sql = "CREATE TABLE students
+    (
+      id INTEGER primary key autoincrement,
+      name TEXT,
+      tagline TEXT,
+      blog_url TEXT,
+      twiiter TEXT,
+      image_url TEXT,
+      biography TEXT,
+      github TEXT
+      )"
+    DB[:conn].execute(sql)
+  end
+```
+
+## RSpec Test 3: `::drop_table`
+
+```
+Student
+  ::drop_table
+     drops the student table (FAILED - 1)
+
+Failures
+  1) Student::drop_table drops the student table
+     Failure/Error: Student.drop_table
+     NoMethodError:
+       undefined method `drop_table' for Student:Class
+     # ./spec/student_spec.rb:55:in `block (3 levels) in <top (required)>'
+
+```
+
+To pass this test, we need to write a `drop_table` class method that will drop the student table.
+
+```ruby
+ def self.drop_table
+     sql = "DROP TABLE students"
+    DB[:conn].execute(sql)
+  end
+```
+
+## RSpec Test 4: `#inserts`
+
+```
+Student
+  #inserts
+     inserts the student into the database (FAILED - 1)
+
+Failures:
+  1) Student#insert inserts the student into the database
+     Failure/Error: avi.insert
+     NoMethodError:
+       undefined method `insert' for #<Student:0x007fb44947db28>
+     # ./spec/student_spec.rb:73:in `block (3 levels) in <top (required)>'
+
+```
+
+This time we need to define an instance method called `inserts` that will insert a new student instance into the student database.  To get this test to pass, we will write out a string verstion of the SQL command for insert into, establish a connection to our application database and execute the sql command.
```

Comment:
Maybe first show how you would do INSERTs without all the variables, then point out that there is a common theme there! and then code it up